lib/routes/workflows.js

```diff
@@ -32,108 +33,109 @@ const routes = application => {
      * @param {object} next - Express callback
     */
     app.get(`/${config.content.base}/:type/:id/:revision/${config.content.actions.approve}`, (req, res, next) => {
-      const type = utils.singleItem('id', req.params.type.toLowerCase(), types);
-      const workflow = workflows.workflow(type, allflows, config, req);
-      const errors = _.get(req.session, `form.content.approve[${req.params.type.toLowerCase()}].errors`, {});
-      const values = _.get(req.session, `form.content.approve[${req.params.type.toLowerCase()}].content`, {});
-
-      // check type exists
-      if (!utils.content.check.type(req, type)) {
-        return next();
-      }
-
-      // id in url is not UUID
-      if (!utils.content.check.id(req)) {
-        return next();
-      }
-
-      // revision in url is not a number
-      if (!utils.content.check.revision(req)) {
-        return next();
-      }
-
-      // check workflow exists
-      if (!workflow) {
-        return next();
-      }
-
-      const steps = (JSON.parse(JSON.stringify(workflow.steps))).reverse();
-
-      _.unset(req.session, 'form.content.approve');
+      utils.fill(types, req.params.type.toLowerCase(), references, database).then(type => {
```

Comment:
actually, we could stick with the original way to get `type`, (`utils.singleItem...`) then, for `utils.fill`, you could send in `type` instead of `types`. That would clean up some error checking on fill and remove the need for `ct.index`